src/index.d.ts

```diff
@@ -0,0 +1,125 @@
+declare module "hyperapp" {
+  // A Hyperapp application instance has an initial state and a base view.
+  // It must also be mounted over an available DOM element.
+  type App<S, P, D> = Readonly<{
+    init: Action<S, P, D>;
```

Comment:
It seems that `init` can take the above as well as `S`

Reply:
Yes, are you referring to line 44 ?
https://github.com/jorgebucaran/hyperapp/pull/954/files/b36ac7fb6a21fdf340ecebb6ae1290cb50871cad#diff-b739f95ace6bbf2f0fd5e8cf1dab97d8R44

Reply:
My apologies, yes.  I think that the name Action for that Union is not really intuitive.  Can you really provide a state object to a event handler?

Reply:
@gamebox Yes, you can, but we don't want people to do that, so is it possible to have TypeScript complain if you try that?

Reply:
We just need to adjust the types so that Action<...> for event handlers excludes that case.

Reply:
Just to be clear, something like
```js
h("button", { onclick: "Mario Mario" }, "It's a me, Mario!")
```
should be frowned upon, correct ?

Reply:
@icylace I believe that is correct.

@jorgebucaran It also seems that the first type in the Action<..> union doesn't make much sense for init either:

`(state: State<S>, props?: Payload<P>) => Action<S, P, D>`

Reply:
@gamebox I've made some updates and put them into a new pull request:  https://github.com/jorgebucaran/hyperapp/pull/969